js/stdlib/ts/ALGO.ts

```diff
@@ -2189,18 +2217,21 @@ const verifyContract_ = async (label:string, info: ContractInfo, bin: Backend, e
     const es = j2sf(e);
     chk(cmp(a, e), `${msg}: expected ${es}, got ${as}`);
   };
-  const chkeq_b = chkeq_x<boolean>((a:boolean, b:boolean) => a === b);
   const chkeq_bn_ = chkeq_x<BigNumber>((a:BigNumber, b:BigNumber) => a.eq(b));
   const chkeq_bn = (a:unknown, b:unknown, msg:string) => chkeq_bn_(bigNumberify(a), bigNumberify(b), msg);
   type Program = string|undefined;
   const chkeq_bs = chkeq_x<Program>((a:Program, b:Program) => j2sf(a) === j2sf(b));
 
-  const appInfoM = await getApplicationInfoM(ApplicationID);
-  if ( 'exn' in appInfoM ) {
+  const appInfoM_live = await getApplicationInfoM(ApplicationID);
+  const appInfoM = 'val' in appInfoM_live ? appInfoM_live
+                                          : await getDeletedApplicationInfoM(ApplicationID);
```

Comment:
You should put all of this logic just in `getApplicationInfoM`... it should just do this

Reply:
And, you should specifically look for the error that says "It doesn't exist" or "It was deleted" or whatever it is

Reply:
I thought, by the way, that you said that when you query the indexer, you can get it to tell you about deleted applications, but that it didn't have enough information. I think the flow is: Ask node, it says no, ask indexer, it says it exists but was deleted, then you go ask the indexer for the transactions so you can find the created one, and then synthesize the response